### DISCUSSION TITLE
useEffect causes internal error undefined

### DISCUSSION THREAD

**Discussion by martinowren** [vercel/next.js#14064] [Help] [answered]
Hi!
I am getting Deployment has failed due to an internal error. (code: undefined) from vercel when I try to push my code to their environment (Works locally)
This is the code, the gallery is there, but did not take it with me for this discussion. 
`import Customers from '../../dist/json/customers';
import { useRouter } from 'next/router';
import { useState, useEffect } from 'react';
import { PortfolioItem } from '../../components/Portfolio';
import { AnimateUp } from '../../components/Animations';

const PortfolioPost = () => {
	const router = useRouter();
	const { slug } = router.query;

	const CustomerArr = Customers;
	const currentCustomer = CustomerArr.find(
		(element) => element.slug === slug
	);
	const [portfolioItem, setPortfolioItem] = useState(null);
	useEffect(() => {
		setPortfolioItem(currentCustomer);
	}, [currentCustomer]);

	return (
		<>
			{portfolioItem && (
				<>
					<PortfolioItem isPurple portofolioObject={portfolioItem} />
					<Gallery
						title={portfolioItem.subtitle}
						text={portfolioItem.descriptionShort}
						feedback={portfolioItem.feedback}
						feedbackAuthor={portfolioItem.feedbackAuthor}
						feedbackImg={portfolioItem.feedbackImg}
						gallery={portfolioItem.imgArray}
					/>
				</>
			)}
		</>
	);
};`

**Comment by jamesmosier** [comment]
Not sure if this will help, but shouldn't you be able to set `currentCustomer` as the default state?

```js
const CustomerArr = Customers;
const currentCustomer = CustomerArr.find(
  (element) => element.slug === slug
);
const [portfolioItem, setPortfolioItem] = useState(currentCustomer);
```

assuming `Customers` is not coming from some async data source. 

I'm guessing that `useEffect` is running on every render because a different object (from a referential equality standpoint) will be returned on each render as you run this:

```
const currentCustomer = CustomerArr.find(
  (element) => element.slug === slug
);
// currentCustomer is "different" on every render
```

but i'm not certain why that would be failing a build.

**Comment by martinowren** [comment]
Hi!
When I set currentCustomer to default state, most times there would not be any data in the state when it rendered. That is why I had to go over to useEffect. 

The error only happened after I added useEffect, so it has to be something with it. 
![image](https://user-images.githubusercontent.com/22822433/84404317-f6494680-ac06-11ea-9d06-ad47e03f0900.png)

**Reply by jamesmosier** [reply]
It's hard to say without seeing all that code that leads up to this component. Try adding `console.log` in the `useEffect` see what is output? Only call `setPortfolioItem` if `currentCustomer` has a value?

**Comment by martinowren** [comment]
I literally can't find the bug. I've commented out everything in that comment and I cannot find it. If I share the repo with you, do you think you can take a look?

**Comment by martinowren** [comment, marked as answer]
Found the issue, someone of the other people in the repo had changed the repo name. Everything working now.

**Reply by fikrikarim** [reply]
Dude I got the exact same problem. Thanks for writing the solution here 👍
